File: JavaScript/Array/L_rangeSumOfSortedSubarr.js
var rangeSum = function (nums, n, left, right) {
    const MOD = 1000000007;

    // Find the maximum possible sum
    const maxSum = nums.reduce((a, b) => a + b, 0);

    // Count the frequency of each sum
    const count = new Array(maxSum + 1).fill(0);
    for (let i = 0; i < n; i++) {
        let sum = 0;
        for (let j = i; j < n; j++) {
            sum += nums[j];
            count[sum]++;
        }
    }

    // Calculate prefix sum of frequencies
    for (let i = 1; i <= maxSum; i++) {
        count[i] += count[i - 1];
    }

    let result = 0;
    let k = 1;
    for (let sum = 1; sum <= maxSum; sum++) {
        while (k <= count[sum]) {
            if (k >= left && k <= right) {
                result = (result + sum) % MOD;
            }
            k++;
        }
    }

    return result;
};

File: Python/Arrays/L_rangeSumOfSortedSubarr.py
def rangeSum(nums, n: int, left: int, right: int) -> int:
    all_sum = []
    for i in range(n):
        curr_sum = 0
        for j in range(i,n):
            curr_sum += nums[j]
            all_sum.append(curr_sum)
    all_sum.sort()
    return sum(all_sum[left-1:right])

print(rangeSum([1,2,3,4], 4, 1, 5))